### PR TITLE
API & client improvements

### PR DESCRIPTION
**Pull request by pfrazee**
Updates:

- Adds typeguard helpers to generated code (eg `isRecord(v: unknown): v is Record`)
- Adds validation helpers to generated code (eg `validateRecord(v: unknown): ValidationResult`)
- Adds a general `validate()` method to the `Lexicons` class.
- Adds a `validateResponse` option to xrpc-server which can be set to `false` to disable validation of responses
- Adds client-side validation of XRPC responses along with a new `XRPCInvalidResponseError` for failed validation.

Basically this PR is tackling a set of tasks we've had queued up in this space. The type-guard and validation helpers are something I've been needing in the app. The client-side validation is to begin to address issues with "faulty" servers.

### CODE REVIEW

File: packages/lexicon/src/types.ts [hunk 1]
```diff
@@ -278,7 +278,7 @@ export function isValidLexiconDoc(v: unknown): v is LexiconDoc {
 }
 
 export function isObj(obj: unknown): obj is Record<string, unknown> {
-  return !!obj && typeof obj === 'object'
+  return obj !== null && typeof obj === 'object'
```
Comment (dholms) [inline packages/lexicon/src/types.ts:281]: worth throwing in any extra checks like `!Array.isArray()` here?

Comment (pfrazee) [reply packages/lexicon/src/types.ts:281]: The way I use it currently, there's pretty much always a `hasProp()` check after, or other validation. Probably not needed?

**Review by dholms — APPROVED** [review]
awesome much needed 🙏

**Review by devinivy — APPROVED** [review]
😎 nice!
